### PR TITLE
[INLONG-11997][Manager] Add comprehensive audit alert rule management API

### LINKED ISSUE #11923: [Umbrella] Tencent Rhino-bird: InLong Audit monitoring capability building
### Describe the proposal

InLong audit is a subsystem independent of InLong, which performs real-time audit and reconciliation of the incoming and outgoing traffic of the Agent, Data Proxy, and Sort modules in the InLong system.

Audit reconciliation is based on the unified caliber of log reporting time, and all services participating in the audit will perform real-time reconciliation according to the same log time. Through audit reconciliation, we can have a clear understanding of the transmission status of each module in InLong, as well as whether there is any loss or duplication of data flow
The current inlong audit is only used for page audit display. Here, we hope to add the ability to increase alarms and monitor based on audit data

For more details on OpenTelemetry, please see https://opentelemetry.io/

### Task list

-

### InLong Component

InLong Audit, InLong Manager

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #11997: [Feature][Manager] Add comprehensive audit alert rule management API
### Description

Currently, Apache InLong lacks a comprehensive audit alert rule management system to monitor data quality, detect anomalies, and provide real-time alerting capabilities. Users need the ability to create, manage, and monitor audit alert rules to ensure data pipeline reliability and quality.

This enhancement adds a complete audit alert rule management API that allows users to:
- Configure alert rules for data count, delay, and quality metrics
- Set up notification channels (EMAIL, SMS, etc.)
- Manage alert rule lifecycle (create, update, delete, enable/disable)
- Query and filter alert rules by group and stream

### Use case

Currently, Apache InLong lacks a comprehensive audit alert rule management system to monitor data quality, detect anomalies, and provide real-time alerting capabilities. Users need the ability to create, manage, and monitor audit alert rules to ensure data pipeline reliability and quality.

This enhancement adds a complete audit alert rule management API that allows users to:
- Configure alert rules for data count, delay, and quality metrics
- Set up notification channels (EMAIL, SMS, etc.)
- Manage alert rule lifecycle (create, update, delete, enable/disable)
- Query and filter alert rules by group and stream

### Are you willing to submit PR?

- [x] Yes, I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by shuideyimei2**
[INLONG-11997][Manager] Add comprehensive audit alert rule management API

Fixes #11997

### Motivation

Currently, Apache InLong lacks a comprehensive audit alert rule management system to monitor data quality, detect anomalies, and provide real-time alerting capabilities. Users need the ability to create, manage, and monitor audit alert rules to ensure data pipeline reliability and quality.

This enhancement adds a complete audit alert rule management API that allows users to:
- Configure alert rules for data count, delay, and quality metrics
- Set up notification channels (EMAIL, SMS, etc.)
- Manage alert rule lifecycle (create, update, delete, enable/disable)
- Query and filter alert rules by group and stream

### Modifications

**1. Data Model Layer**
- Added [AuditAlertRule](file://e:\inlong\inlong-manager\manager-pojo\src\main\java\org\apache\inlong\manager\pojo\audit\AuditAlertRule.java#L25-L63) POJO class in `manager-pojo` module with comprehensive validation annotations
- Added [AuditAlertRuleEntity](file://e:\inlong\inlong-manager\manager-dao\src\main\java\org\apache\inlong\manager\dao\entity\AuditAlertRuleEntity.java#L23-L39) database entity in `manager-dao` module for MyBatis mapping
- Added [AuditAlertRuleEntityMapper](file://e:\inlong\inlong-manager\manager-dao\src\main\java\org\apache\inlong\manager\dao\mapper\AuditAlertRuleEntityMapper.java#L24-L32) interface with full CRUD operations
- Added MyBatis XML mapping file [AuditAlertRuleEntityMapper.xml](file://e:\inlong\inlong-manager\manager-dao\src\main\resources\mappers\AuditAlertRuleEntityMapper.xml) with optimized SQL queries

**2. Service Layer**
- Extended [AuditService](file://e:\inlong\inlong-manager\manager-service\src\main\java\org\apache\inlong\manager\service\core\AuditService.java#L31-L103) interface with audit alert rule management methods
- Implemented audit alert rule management in [AuditServiceImpl](file://e:\inlong\inlong-manager\manager-service\src\main\java\org\apache\inlong\manager\service\core\impl\AuditServiceImpl.java#L78-L446) class:
  - [createAlertRule()](file://e:\inlong\inlong-manager\manager-service\src\main\java\org\apache\inlong\manager\service\core\AuditService.java#L87-L87) - Create new alert rules with validation
  - [getAlertRule()](file://e:\inlong\inlong-manager\manager-service\src\main\java\org\apache\inlong\manager\service\core\AuditService.java#L92-L92) - Retrieve alert rule by ID
  - [updateAlertRule()](file://e:\inlong\inlong-manager\manager-service\src\main\java\org\apache\inlong\manager\service\core\AuditService.java#L97-L97) - Update existing alert rules
  - [deleteAlertRule()](file://e:\inlong\inlong-manager\manager-service\src\main\java\org\apache\inlong\manager\service\core\AuditService.java#L102-L102) - Delete alert rules
  - [listAlertRules()](file://e:\inlong\inlong-manager\manager-service\src\main\java\org\apache\inlong\manager\service\core\AuditService.java#L82-L82) - Query alert rules with filtering support

**3. Web Layer**
- Added REST API endpoints in [AuditController](file://e:\inlong\inlong-manager\manager-web\src\main\java\org\apache\inlong\manager\web\controller\AuditController.java#L51-L137):
  - `POST /api/audit/alert/rule` - Create alert rule
  - `GET /api/audit/alert/rule/{id}` - Get alert rule by ID
  - `PUT /api/audit/alert/rule` - Update alert rule
  - `DELETE /api/audit/alert/rule/{id}` - Delete alert rule
  - `GET /api/audit/alert/rule/list` - List alert rules with filtering
  - `GET /api/audit/alert/rule/enabled` - Get all enabled alert rules

**4. Database Schema**
- Database table `audit_alert_rule` with optimized indexes
- Support for MySQL with proper charset and collation

**5. Testing**
- Comprehensive unit tests in [AuditAlertRuleTest](file://e:\inlong\inlong-manager\manager-service\src\test\java\org\apache\inlong\manager\service\core\AuditAlertRuleTest.java#L34-L152) class
- API integration test scripts for manual validation
- Database initialization SQL scripts

**6. Configuration**
- MyBatis configuration for audit alert rule mapping
- Spring Boot auto-configuration support

### Verifying this change

- [x] This change added tests and can be verified as follows:

**Unit Tests:**
- Added [AuditAlertRuleTest](file://e:\inlong\inlong-manager\manager-service\src\test\java\org\apache\inlong\manager\service\core\AuditAlertRuleTest.java#L34-L152) class with comprehensive test coverage:
  - Test alert rule creation with validation
  - Test alert rule retrieval and updates
  - Test alert rule deletion and lifecycle management
  - Test query operations with filtering
  - Test error handling and edge cases

**Integration Tests:**
- Added API test scripts ([test-audit-alert-api.sh](file://e:\inlong\inlong-manager\test-audit-alert-api.sh) and [test-audit-alert-api.ps1](file://e:\inlong\inlong-manager\test-audit-alert-api.ps1))
- Added database initialization script ([init-audit-alert-test-data.sql](file://e:\inlong\inlong-manager\init-audit-alert-test-data.sql))
- Manual testing workflow documented

**Test Coverage:**
- Service layer: Full CRUD operations testing
- Controller layer: HTTP endpoint testing with various scenarios
- Data layer: MyBatis mapping and SQL query validation
- Error handling: Input validation and exception scenarios

### CODE REVIEW

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/AuditClient.java [hunk 1]
```diff
@@ -83,4 +85,92 @@ public Boolean refreshCache(AuditRequest request) {
...
+
+    // Audit Alert Rule methods
+
+    /**
+     * Create an audit alert rule
+     *
+     * @param request The audit alert rule to create
+     * @return The created audit alert rule ID
+     */
+    public Integer createAlertRule(AuditAlertRuleRequest request) {
```
Comment (fuweng11) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/AuditClient.java:97]:
```suggestion
    public Integer create(AuditAlertRuleRequest request) {
```

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/AuditClient.java [hunk 2]
```diff
@@ -83,4 +85,92 @@ public Boolean refreshCache(AuditRequest request) {
...
+        return response.getData();
+    }
+
+    /**
+     * Get an audit alert rule by ID
+     *
+     * @param id The rule ID
+     * @return The audit alert rule
+     */
+    public AuditAlertRule getAlertRule(Integer id) {
```
Comment (fuweng11) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/AuditClient.java:118]:
```suggestion
    public AuditAlertRule get(Integer id) {
```

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/Condition.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    private String operator;
+
+    @ApiModelProperty(value = "Value for comparison", required = true)
+    @NotNull(message = "Value cannot be null")
+    private Object value;
+
+    public Condition() {
+    }
+
+    public Condition(String type, String operator, Object value) {
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/Condition.java:46]: After the @Data annotation, the set, get, and other methods for all the properties in the class will be automatically implemented.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/Condition.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        this.type = type;
+        this.operator = operator;
+        this.value = value;
+    }
+
+    @Override
+    public String toString() {
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/Condition.java:53]: Ditto.

File: inlong-manager/manager-dao/src/main/resources/mappers/AuditAlertRuleEntityMapper.xml [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            </if>
+        </where>
+        order by create_time desc
+    </select>
+
+    <select id="selectEnabledRules" resultMap="BaseResultMap">
+        select
+        <include refid="Base_Column_List"/>
+        from audit_alert_rule
+        where enabled = 1 and is_deleted = 0
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/resources/mappers/AuditAlertRuleEntityMapper.xml:138]:
```suggestion
        where enabled = 1 
        and is_deleted = 0
```

File: inlong-manager/manager-client/src/test/java/org/apache/inlong/manager/client/api/integration/AuditAlertRuleIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+                        .willReturn(okJson(responseBody)));
+
+        // Execute test
+        Boolean result = auditClient.delete(createdRuleId);
+
+        // Verify result
+        Assertions.assertNotNull(result, "Delete result should not be null");
+        Assertions.assertTrue(result, "Delete should be successful");
+
+        System.out.println("✓ Successfully deleted alert rule, ID: " + createdRuleId);
```
Comment (fuweng11) [inline inlong-manager/manager-client/src/test/java/org/apache/inlong/manager/client/api/integration/AuditAlertRuleIntegrationTest.java:331]: Ditto.

**Review by fuweng11 — APPROVED** [review]

**Review by doleyzi — APPROVED** [review]
